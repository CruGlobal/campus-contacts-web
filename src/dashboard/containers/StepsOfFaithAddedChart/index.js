import React, { useContext } from 'react';
import { useTranslation } from 'react-i18next';
import { useQuery } from 'react-apollo-hooks';
import gql from 'graphql-tag';
import styled from '@emotion/styled';
import moment from 'moment';

import BarChart from '../../components/BarChart';
import AppContext from '../../appContext';

const GET_STAGES_REPORT = gql`
    query organizationPathwayStagesReport(
        $period: String!
        $organizationId: ID!
        $endDate: ISO8601DateTime!
    ) {
        organizationPathwayStagesReport(
            period: $period
            organizationId: $organizationId
            endDate: $endDate
        ) {
            memberCount
            pathwayStage {
                name
            }
            othersStepsAddedCount
        }
    }
`;

const Wrapper = styled.div`
    height: 400px;
`;

const StepsOfFaithAddedChart = () => {
    const { orgId } = useContext(AppContext);

    const { data, loading } = useQuery(GET_STAGES_REPORT, {
        variables: {
            period: '',
            organizationId: orgId,
            endDate: moment().format(),
        },
    });
    const { t } = useTranslation('insights');

    if (loading) {
        return <Wrapper>{t('loading')}</Wrapper>;
    }

    const { organizationPathwayStagesReport: report } = data;

    const LABEL = t('stepsOfFaith.legendLabel');
    const STAGE_LABEL = t('stage');

    const graphData = report.map(row => ({
        [LABEL]: row.othersStepsAddedCount,
        [STAGE_LABEL]: row.pathwayStage.name.toUpperCase(),
    }));

    return (
        <Wrapper>
            <BarChart data={graphData} keys={[LABEL]} indexBy={STAGE_LABEL} />
        </Wrapper>
    );
};

export default StepsOfFaithAddedChart;
