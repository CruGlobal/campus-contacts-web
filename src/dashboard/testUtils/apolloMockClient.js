import { ApolloClient } from 'apollo-client';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { SchemaLink } from 'apollo-link-schema';
import { addMockFunctionsToSchema } from 'graphql-tools';
import { buildClientSchema } from 'graphql/utilities';
import { createMockerFromIntrospection } from 'fraql/mock';
import gql from 'fraql';

import introspectionQuery from '../../../schema.json';

import { globalMocks } from './globalMocks';

export const createApolloMockClient = (mocks = {}) => {
    const schema = buildClientSchema(introspectionQuery);

    addMockFunctionsToSchema({
        schema,
        mocks: { ...globalMocks, ...mocks },
    });

    return new ApolloClient({
        cache: new InMemoryCache(),
        link: new SchemaLink({ schema }),
    });
};

export const mockFragment = (fragmentDocument, options) => {
    return createMockerFromIntrospection(introspectionQuery, {
        mocks: globalMocks,
    }).mockFragment(gql(fragmentDocument.loc.source.body), options);
};
