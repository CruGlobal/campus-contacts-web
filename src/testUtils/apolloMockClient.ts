import { ApolloClient } from 'apollo-client';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { SchemaLink } from 'apollo-link-schema';
import { addMockFunctionsToSchema } from 'graphql-tools';
// @ts-ignore
import { buildClientSchema } from 'graphql/utilities';
// @ts-ignore
import { createMockerFromIntrospection } from 'fraql/mock';
// @ts-ignore
import gql from 'fraql';

import introspectionQuery from '../../schema.json';

import { globalMocks } from './globalMocks';

export const createApolloMockClient = (mocks = {}) => {
    const schema = buildClientSchema(introspectionQuery);

    addMockFunctionsToSchema({
        schema,
        mocks: { ...globalMocks, ...mocks },
    });

    return new ApolloClient({
        cache: new InMemoryCache(),
        link: new SchemaLink({ schema }),
    });
};

export const mockFragment = (fragmentDocument: any, options: any) => {
    return createMockerFromIntrospection(introspectionQuery, {
        mocks: globalMocks,
    }).mockFragment(gql(fragmentDocument.loc.source.body), options);
};
