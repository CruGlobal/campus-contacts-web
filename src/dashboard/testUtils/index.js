import React, { ReactElement } from 'react';
import { ApolloProvider } from 'react-apollo-hooks';
import { render } from '@testing-library/react';

import { createApolloMockClient } from './apolloMockClient';
import defaultTheme from '../defaultTheme';
import { ThemeProvider } from 'emotion-theming';
import AppContext from '../appContext';

export const renderWithContext = (component, { mocks } = {}, context = {}) => {
    const mockApolloClient = createApolloMockClient(mocks);

    // Warning: don't call any functions in here that return new instances on every call.
    // All the props need to stay the same otherwise renderer won't work.
    const wrapper = ({ children }) => (
        <ApolloProvider client={mockApolloClient}>
            <AppContext.Provider value={context}>
                <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
            </AppContext.Provider>
        </ApolloProvider>
    );

    const renderResult = render(component, { wrapper });

    return {
        ...renderResult,
        snapshot: () => {
            expect(renderResult.container).toMatchSnapshot();
        },
    };
};
