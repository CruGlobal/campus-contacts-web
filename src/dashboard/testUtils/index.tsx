import React, { ReactNode } from 'react';
import { ApolloProvider } from 'react-apollo-hooks';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'emotion-theming';

import defaultTheme from '../defaultTheme';
import AppContext from '../appContext';

import { createApolloMockClient } from './apolloMockClient';

export const renderWithContext = (
    component: any,
    { mocks: mocks = {}, appContext = {} } = {},
) => {
    const mockApolloClient = createApolloMockClient(mocks);

    // Warning: don't call any functions in here that return new instances on every call.
    // All the props need to stay the same otherwise renderer won't work.
    const wrapper = ({ children }: { children: ReactNode }) => (
        <ApolloProvider client={mockApolloClient}>
            <AppContext.Provider value={appContext}>
                <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
            </AppContext.Provider>
        </ApolloProvider>
    );

    // @ts-ignore
    const renderResult = render(component, { wrapper });
    return {
        ...renderResult,
        snapshot: () => {
            expect(renderResult.container).toMatchSnapshot();
        },
    };
};
