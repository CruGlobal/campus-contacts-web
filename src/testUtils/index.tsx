import React, { ReactNode, ReactElement } from 'react';
import { ApolloProvider } from 'react-apollo-hooks';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'emotion-theming';
import { IMocks } from 'graphql-tools';

import defaultTheme from '../defaultTheme';
import {
    AppContext,
    AppContextValue,
    appContextDefaultValue,
} from '../appContext';

import { createApolloMockClient } from './apolloMockClient';

interface RenderWithContextParams {
    mocks?: IMocks;
    appContext?: AppContextValue;
}

export const renderWithContext = (
    component: ReactElement,
    {
        mocks: mocks = {},
        appContext = appContextDefaultValue,
    }: RenderWithContextParams = {},
) => {
    const mockApolloClient = createApolloMockClient(mocks);

    // Warning: don't call any functions in here that return new instances on every call.
    // All the props need to stay the same otherwise renderer won't work.
    const wrapper = ({ children }: { children?: ReactNode }) => (
        <ApolloProvider client={mockApolloClient}>
            <AppContext.Provider value={appContext}>
                <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
            </AppContext.Provider>
        </ApolloProvider>
    );

    const renderResult = render(component, { wrapper });
    return {
        ...renderResult,
        snapshot: () => {
            expect(renderResult.container).toMatchSnapshot();
        },
    };
};
