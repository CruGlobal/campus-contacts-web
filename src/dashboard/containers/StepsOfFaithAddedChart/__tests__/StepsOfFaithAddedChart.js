// Vendors
import React from 'react';

import { renderWithContext } from '../../../testUtils';
import StepsOfFaithAddedChart from '../';

describe('<StepsOfFaithAddedChart />', () => {
    it('should render properly loading state', async () => {
        renderWithContext(<StepsOfFaithAddedChart />).snapshot();
    });
});
