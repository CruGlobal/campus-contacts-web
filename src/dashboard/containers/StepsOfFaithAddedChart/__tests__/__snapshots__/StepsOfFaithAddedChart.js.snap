// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StepsOfFaithAddedChart /> should render properly loading state 1`] = `
<div>
  <div
    class="css-15l45ju"
  >
    Loading...
  </div>
</div>
`;

exports[`<PersonalStepsMemberStagesChart /> should render properly with chart 1`] = `
<div>
  <div
    class="css-15l45ju"
  >
    <div
      style="position: relative;"
    >
      <svg
        height="400"
        role="img"
        width="848"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="transparent"
          height="400"
          width="848"
        />
        <g
          transform="translate(30,30)"
        >
          <g>
            <line
              opacity="1"
              stroke="#ECEEF2"
              stroke-width="1"
              x1="0"
              x2="818"
              y1="280"
              y2="280"
            />
            <line
              opacity="1"
              stroke="#ECEEF2"
              stroke-width="1"
              x1="0"
              x2="818"
              y1="255"
              y2="255"
            />
            <line
              opacity="1"
              stroke="#ECEEF2"
              stroke-width="1"
              x1="0"
              x2="818"
              y1="229"
              y2="229"
            />
            <line
              opacity="1"
              stroke="#ECEEF2"
              stroke-width="1"
              x1="0"
              x2="818"
              y1="204"
              y2="204"
            />
            <line
              opacity="1"
              stroke="#ECEEF2"
              stroke-width="1"
              x1="0"
              x2="818"
              y1="178"
              y2="178"
            />
            <line
              opacity="1"
              stroke="#ECEEF2"
              stroke-width="1"
              x1="0"
              x2="818"
              y1="153"
              y2="153"
            />
            <line
              opacity="1"
              stroke="#ECEEF2"
              stroke-width="1"
              x1="0"
              x2="818"
              y1="127"
              y2="127"
            />
            <line
              opacity="1"
              stroke="#ECEEF2"
              stroke-width="1"
              x1="0"
              x2="818"
              y1="102"
              y2="102"
            />
            <line
              opacity="1"
              stroke="#ECEEF2"
              stroke-width="1"
              x1="0"
              x2="818"
              y1="76"
              y2="76"
            />
            <line
              opacity="1"
              stroke="#ECEEF2"
              stroke-width="1"
              x1="0"
              x2="818"
              y1="51"
              y2="51"
            />
            <line
              opacity="1"
              stroke="#ECEEF2"
              stroke-width="1"
              x1="0"
              x2="818"
              y1="25"
              y2="25"
            />
            <line
              opacity="1"
              stroke="#ECEEF2"
              stroke-width="1"
              x1="0"
              x2="818"
              y1="0"
              y2="0"
            />
          </g>
          <g
            transform="translate(0,280)"
          >
            <g
              style="opacity: 1;"
              transform="translate(239,0)"
            >
              <line
                style="stroke: #777777; stroke-width: 1;"
                x1="0"
                x2="0"
                y1="0"
                y2="0"
              />
              <text
                dominant-baseline="text-before-edge"
                style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                text-anchor="middle"
                transform="translate(0,20) rotate(0)"
              >
                NO STAGE
              </text>
            </g>
            <g
              style="opacity: 1;"
              transform="translate(579,0)"
            >
              <line
                style="stroke: #777777; stroke-width: 1;"
                x1="0"
                x2="0"
                y1="0"
                y2="0"
              />
              <text
                dominant-baseline="text-before-edge"
                style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                text-anchor="middle"
                transform="translate(0,20) rotate(0)"
              >
                UNINTERESTED
              </text>
            </g>
            <line
              style="stroke: transparent; stroke-width: 1;"
              x1="0"
              x2="818"
              y1="0"
              y2="0"
            />
          </g>
          <g
            transform="translate(0,0)"
          >
            <g
              style="opacity: 1;"
              transform="translate(0,280)"
            >
              <line
                style="stroke: #777777; stroke-width: 1;"
                x1="0"
                x2="0"
                y1="0"
                y2="0"
              />
              <text
                dominant-baseline="central"
                style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                text-anchor="end"
                transform="translate(-5,0) rotate(0)"
              >
                0
              </text>
            </g>
            <g
              style="opacity: 1;"
              transform="translate(0,255)"
            >
              <line
                style="stroke: #777777; stroke-width: 1;"
                x1="0"
                x2="0"
                y1="0"
                y2="0"
              />
              <text
                dominant-baseline="central"
                style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                text-anchor="end"
                transform="translate(-5,0) rotate(0)"
              >
                1
              </text>
            </g>
            <g
              style="opacity: 1;"
              transform="translate(0,229)"
            >
              <line
                style="stroke: #777777; stroke-width: 1;"
                x1="0"
                x2="0"
                y1="0"
                y2="0"
              />
              <text
                dominant-baseline="central"
                style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                text-anchor="end"
                transform="translate(-5,0) rotate(0)"
              >
                2
              </text>
            </g>
            <g
              style="opacity: 1;"
              transform="translate(0,204)"
            >
              <line
                style="stroke: #777777; stroke-width: 1;"
                x1="0"
                x2="0"
                y1="0"
                y2="0"
              />
              <text
                dominant-baseline="central"
                style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                text-anchor="end"
                transform="translate(-5,0) rotate(0)"
              >
                3
              </text>
            </g>
            <g
              style="opacity: 1;"
              transform="translate(0,178)"
            >
              <line
                style="stroke: #777777; stroke-width: 1;"
                x1="0"
                x2="0"
                y1="0"
                y2="0"
              />
              <text
                dominant-baseline="central"
                style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                text-anchor="end"
                transform="translate(-5,0) rotate(0)"
              >
                4
              </text>
            </g>
            <g
              style="opacity: 1;"
              transform="translate(0,153)"
            >
              <line
                style="stroke: #777777; stroke-width: 1;"
                x1="0"
                x2="0"
                y1="0"
                y2="0"
              />
              <text
                dominant-baseline="central"
                style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                text-anchor="end"
                transform="translate(-5,0) rotate(0)"
              >
                5
              </text>
            </g>
            <g
              style="opacity: 1;"
              transform="translate(0,127)"
            >
              <line
                style="stroke: #777777; stroke-width: 1;"
                x1="0"
                x2="0"
                y1="0"
                y2="0"
              />
              <text
                dominant-baseline="central"
                style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                text-anchor="end"
                transform="translate(-5,0) rotate(0)"
              >
                6
              </text>
            </g>
            <g
              style="opacity: 1;"
              transform="translate(0,102)"
            >
              <line
                style="stroke: #777777; stroke-width: 1;"
                x1="0"
                x2="0"
                y1="0"
                y2="0"
              />
              <text
                dominant-baseline="central"
                style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                text-anchor="end"
                transform="translate(-5,0) rotate(0)"
              >
                7
              </text>
            </g>
            <g
              style="opacity: 1;"
              transform="translate(0,76)"
            >
              <line
                style="stroke: #777777; stroke-width: 1;"
                x1="0"
                x2="0"
                y1="0"
                y2="0"
              />
              <text
                dominant-baseline="central"
                style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                text-anchor="end"
                transform="translate(-5,0) rotate(0)"
              >
                8
              </text>
            </g>
            <g
              style="opacity: 1;"
              transform="translate(0,51)"
            >
              <line
                style="stroke: #777777; stroke-width: 1;"
                x1="0"
                x2="0"
                y1="0"
                y2="0"
              />
              <text
                dominant-baseline="central"
                style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                text-anchor="end"
                transform="translate(-5,0) rotate(0)"
              >
                9
              </text>
            </g>
            <g
              style="opacity: 1;"
              transform="translate(0,25)"
            >
              <line
                style="stroke: #777777; stroke-width: 1;"
                x1="0"
                x2="0"
                y1="0"
                y2="0"
              />
              <text
                dominant-baseline="central"
                style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                text-anchor="end"
                transform="translate(-5,0) rotate(0)"
              >
                10
              </text>
            </g>
            <g
              style="opacity: 1;"
              transform="translate(0,0)"
            >
              <line
                style="stroke: #777777; stroke-width: 1;"
                x1="0"
                x2="0"
                y1="0"
                y2="0"
              />
              <text
                dominant-baseline="central"
                style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                text-anchor="end"
                transform="translate(-5,0) rotate(0)"
              >
                11
              </text>
            </g>
            <line
              style="stroke: transparent; stroke-width: 1;"
              x1="0"
              x2="0"
              y1="0"
              y2="280"
            />
          </g>
          <g>
            <g
              transform="translate(137, 25)"
            >
              <rect
                fill="#3CC8E6"
                height="255"
                rx="0"
                ry="0"
                stroke="#3CC8E6"
                stroke-width="0"
                width="204"
              />
            </g>
            <g
              transform="translate(477, 0)"
            >
              <rect
                fill="#3CC8E6"
                height="280"
                rx="0"
                ry="0"
                stroke="#3CC8E6"
                stroke-width="0"
                width="204"
              />
            </g>
          </g>
        </g>
      </svg>
    </div>
  </div>
</div>
`;
